gcovr/gcov.py

```diff
@@ -75,10 +76,7 @@ def get_datafiles(flist, options):
 
 
 def is_non_code(code):
-    if sys.version_info < (3, 0):
-        code = code.strip().translate(None, '}{')
-    else:
-        code = code.strip().translate(noncode_mapper)
```

Comment:
I was a bit confused by this change, but apparently `unicode.translate()` from Python 2.7 behaves like `str.translate()` in Python 3. It might be good to make that requirement explicit, e.g. by adding a `six` dependency and:

``` python
import six
...

def is_non_code(code):
    assert isinstance(code, six.text_type)
    ...
```

Reply:
[io.open](https://docs.python.org/2/library/io.html#io.open) returns unicode on python2 and bytes on python3

```
Python distinguishes between files opened in binary and text modes, even when the underlying operating system doesn’t. Files opened in binary mode (including 'b' in the mode argument) return contents as bytes objects without any decoding. In text mode (the default, or when 't' is included in the mode argument), the contents of the file are returned as unicode strings, the bytes having been first decoded using a platform-dependent encoding or using the specified encoding if given.
```

Reply:
As we can ensure io.open return unicode, adding additional module `six` is not necessary in my opinion.